### ISSUE TITLE
View.Checkbox calls `checkedChanged` when ListView updates

### ISSUE THREAD

**Issue by Xatter** [fsprojects/Fabulous#519]
I am trying to create a Todo list style example app with Fabulous using the new Checkbox control in 0.41.0. 

```
open System
open System.Diagnostics

open Fabulous
open Fabulous.XamarinForms
open Xamarin.Forms

module App = 
    type Todo = { 
        id : Guid
        title : string
        completed: bool
    }
    with static member Zero = {
            id = Guid.Empty
            title = String.Empty
            completed = false
        }

    type Model = {
        newItemTitle : string
        todos: Todo list
    }
    with static member Zero = {
            newItemTitle = String.Empty
            todos = [
                {id=Guid.NewGuid(); title="Item3"; completed=false}
                {id=Guid.NewGuid(); title="Item2"; completed=false}
                {id=Guid.NewGuid(); title="Item1"; completed=false}
            ]
        }

    type Msg = 
        | Add of Todo
        | NewItemTitle of string
        | ClearTitle
        | ToggleCompleted of Guid * Boolean

    let init () = Model.Zero, Cmd.none

    let update msg model =
        match msg with
        | Add todo -> { model with todos=todo::model.todos }, (Cmd.ofMsg ClearTitle)
        | NewItemTitle title -> { model with newItemTitle=title }, Cmd.none
        | ClearTitle -> { model with newItemTitle=String.Empty }, Cmd.none
        | ToggleCompleted (tid, value) -> 
            let newTodos =
                model.todos
                |> List.map(fun t ->
                    if t.id <> tid then t
                    else
                        { t with completed = value }
                )
            { model with todos=newTodos }, Cmd.none

    let view (model: Model) dispatch =
        View.ContentPage(useSafeArea=true,
            content = View.StackLayout(
                children=[
                    View.StackLayout(
                        orientation=StackOrientation.Horizontal,
                        children=[
                            dependsOn model.newItemTitle (fun model title -> View.Entry(
                                    text=title,
                                    horizontalOptions = LayoutOptions.FillAndExpand,
                                    textChanged = debounce 250 (fun e -> dispatch (NewItemTitle e.NewTextValue))
                            ))
                            View.Button(
                                text = "+",
                                widthRequest=50.0,
                                horizontalOptions=LayoutOptions.End,
                                command = (fun () -> dispatch (Add { Todo.Zero with id=Guid.NewGuid();title=model.newItemTitle }))
                            )
                        ]
                    )

                    dependsOn model.todos (fun model todos -> View.ListView(
                        todos
                        |> List.map(fun item ->
                            View.StackLayout(
                                orientation=StackOrientation.Horizontal,
                                children=[
                                    dependsOn (item.id, item.completed) (fun model (tid, completed) -> 
                                        View.CheckBox(
                                            isChecked=completed,
                                            checkedChanged=(fun e -> dispatch (ToggleCompleted (tid, e.Value)) )
                                        )
                                    )
                                    View.Label( text=item.title )
                                ]
                            )
                        )
                    ))
                ]
            )
        )

    // Note, this declaration is needed if you enable LiveUpdate
    let program = Program.mkProgram init update view

type App () as app = 
    inherit Application ()

    let runner = 
        App.program
#if DEBUG
        |> Program.withConsoleTrace
#endif
        |> XamarinFormsProgram.run app

#if DEBUG
    // Uncomment this line to enable live update in debug mode. 
    // See https://fsprojects.github.io/Fabulous/tools.html for further  instructions.
    //
    //do runner.EnableLiveUpdate()
#endif    

    // Uncomment this code to save the application state to app.Properties using Newtonsoft.Json
    // See https://fsprojects.github.io/Fabulous/models.html for further  instructions.
#if APPSAVE
    let modelId = "model"
    override __.OnSleep() = 

        let json = Newtonsoft.Json.JsonConvert.SerializeObject(runner.CurrentModel)
        Console.WriteLine("OnSleep: saving model into app.Properties, json = {0}", json)

        app.Properties.[modelId] <- json

    override __.OnResume() = 
        Console.WriteLine "OnResume: checking for model in app.Properties"
        try 
            match app.Properties.TryGetValue modelId with
            | true, (:? string as json) -> 

                Console.WriteLine("OnResume: restoring model from app.Properties, json = {0}", json)
                let model = Newtonsoft.Json.JsonConvert.DeserializeObject<App.Model>(json)

                Console.WriteLine("OnResume: restoring model from app.Properties, model = {0}", (sprintf "%0A" model))
                runner.SetCurrentModel (model, Cmd.none)

            | _ -> ()
        with ex -> 
            App.program.onError("Error while restoring model found in app.Properties", ex)

    override this.OnStart() = 
        Console.WriteLine "OnStart: using same logic as OnResume()"
        this.OnResume()
#endif
```

Steps to reproduce:
* check `Item 2` to mark it as `completed`
* add a new todo by typing in a title and pressing the `+` button

From then on it seems the check boxes go wild

This makes it impossible to keep the `completed` value in sync. A possible work around would be to use `click` action instead but this is not exposed on `View.Checkbox`

Is this situation the intent of `View.Stateful` ?

**Comment by TimLariviere** [comment]
It seems to be the same problem than #304 
If you append the todo instead of prepending it, it should be fine.

But in this case, Fabulous tries to reuse the existing rows.
So the "Item2" row becomes "Item3", Fabulous unchecks the row and triggers a CheckedChanged where it shouldn't.
This then creates a chain reaction leading to an invalid state.

I'm currently working on improving this

**Comment by Xatter** [comment]
Yep, sure enough switching this line:
```
       match msg with
        | Add todo -> { model with todos=todo::model.todos }, (Cmd.ofMsg ClearTitle)
```

To:
    `| Add todo -> { model with todos=model.todos@[todo]}, (Cmd.ofMsg ClearTitle)`

Everything works as expected. Just commenting here in case anyone else is running into this

**Comment by TimLariviere** [comment]
The issue regarding the event going wild has been fixed in v0.50.0-alpha.2 (from #530)
The prepending optimization will be taken care of by #454

**Comment by TimLariviere** [comment]
This will be released in v0.50.0 which will arrive in the coming days (once we add support for XF 4.3)
